apps/ui/src/views/Proposal/Overview.vue

```diff
@@ -126,30 +141,38 @@ async function handleCancelClick() {
 }
 
 async function handleAiSummaryClick() {
-  if (aiSummaryBody.value) {
+  if (aiSummaryContent.value) {
     aiSummaryOpen.value = !aiSummaryOpen.value;
     return;
   }
 
+  await fetchAiSummary();
+
+  if (aiSummaryState.value.errored) {
+    return uiStore.addNotification('error', 'There was an error fetching the AI summary.');
+  }
+
+  aiSummaryOpen.value = true;
+}
+
+async function handleAiSpeechClick() {
+  if (['playing', 'paused'].includes(audioState.value)) {
+    return audioState.value === 'playing' ? pauseAudio() : playAudio();
+  }
+
   try {
-    aiSummaryLoading.value = true;
-    const response = await fetch(`https://sh5.co/api/ai/summary/${props.proposal.id}`, {
-      method: 'POST'
-    });
-    const data = await response.json();
-
-    if (data.error) {
-      throw new Error(data.error.message);
-    }
+    await fetchAiSpeech();
```

Comment:
If I play it once and try to play it for second time it will refetch so it's delay introduced there.

We can avoid this by preventing fetch it multiple times, we will also need to adjust useAudio as it consumes array buffer.

Reply:
Added another `state` type in useAudio to avoid refetching the file